From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Samuel Maddock <smaddock@slack-corp.com>
Date: Mon, 6 Oct 2025 16:34:43 -0400
Subject: fix: replace deprecated SetPrototype

https://chromium-review.googlesource.com/c/v8/v8/+/6983465

This is already applied in newer versions of Node so we can drop
this patch once we upgrade.

diff --git a/src/api/environment.cc b/src/api/environment.cc
index d3f02f4b56f035e7fa2a596a029795cd84b71c28..43838863294e954f10bb01f9b2de1b0e9f5ed7af 100644
--- a/src/api/environment.cc
+++ b/src/api/environment.cc
@@ -801,7 +801,7 @@ MaybeLocal<Object> InitializePrivateSymbols(Local<Context> context,
 
   Local<Object> private_symbols_object;
   if (!private_symbols->NewInstance(context).ToLocal(&private_symbols_object) ||
-      private_symbols_object->SetPrototype(context, Null(isolate))
+      private_symbols_object->SetPrototypeV2(context, Null(isolate))
           .IsNothing()) {
     return MaybeLocal<Object>();
   }
@@ -827,7 +827,7 @@ MaybeLocal<Object> InitializePerIsolateSymbols(Local<Context> context,
   Local<Object> per_isolate_symbols_object;
   if (!per_isolate_symbols->NewInstance(context).ToLocal(
           &per_isolate_symbols_object) ||
-      per_isolate_symbols_object->SetPrototype(context, Null(isolate))
+      per_isolate_symbols_object->SetPrototypeV2(context, Null(isolate))
           .IsNothing()) {
     return MaybeLocal<Object>();
   }
